integrations/amazon_bedrock/tests/test_amazon_chat_bedrock.py

```diff
@@ -38,17 +38,12 @@ def mock_prompt_handler():
         yield mock_prompt_handler
 
 
-def test_to_dict(mock_auto_tokenizer, mock_boto3_session):
+def test_to_dict(mock_auto_tokenizer, mock_boto3_session, set_env_variables):
     """
     Test that the to_dict method returns the correct dictionary without aws credentials
     """
     generator = AmazonBedrockChatGenerator(
         model="anthropic.claude-v2",
-        aws_access_key_id="some_fake_id",
-        aws_secret_access_key="some_fake_key",
-        aws_session_token="some_fake_token",
-        aws_profile_name="some_fake_profile",
-        aws_region_name="fake_region",
```

Comment:
Let's actually check for these in the expected dict instead of setting them to `None`. Same with the `from_dict` test.

Reply:
the `to_dict()` method doesn't save any of the AWS env variables/Secret - should it?

Reply:
Ah, I didn't notice that! Yeah, it definitely should - consider a situation where the user passes their own secret (with a different env var). If we don't save it (and load it again), the reconstructed component will try to read from the default env var, which might or might not be populated.

C.f https://github.com/deepset-ai/haystack-core-integrations/pull/386/files

Reply:
OK, I still need to make one more change then: store the variables names in the class instance as parameters. 

Sorry for this confusion. I just assumed that for core-integration components that rely on AWS, nothing is ever stored since everything is standardized (i.e.: en variables names, config files). I would just use the `Secret` component to read the environmental variables and set up the connection, being that the only scope of the variables.

But I will make it look like all the other components.